CMakeLists.txt

```diff
@@ -1,27 +1,44 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(octomap_msgs)
 
-find_package(catkin REQUIRED COMPONENTS message_generation geometry_msgs std_msgs)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+# Disable Wredundant-decls warnings since rosidl generates redundant function declarations 
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wno-redundant-decls")
```

Comment:
Could we use `target_compile_options` or is this not possible due to the use of the `rosidl_generate_interfaces` macro?

Reply:
The redundant declaration warnings are brought by the msgs files generated by the `rosidl_generate_interfaces`, thus they cannot be impacted by `target_compile_options` which only adds compile options to the target created by `add_executable()` or `add_library()`.